api/v1alpha1/resourceflavor_types.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ResourceFlavorSpec defines the desired state of ResourceFlavor
+type ResourceFlavorSpec struct {
+	// resource is the resource name of the flavor, e.g., "cpu".
+	Resource corev1.ResourceName `json:"resource,omitempty"`
+
+	// labels associated with this type. Those labels are matched against or
+	// converted to node affinity constraints on the workload’s pods.
+	// For example, cloud.provider.com/accelerator: nvidia-tesla-k80.
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// taints associated with this constraint that workloads must explicitly
+	// “tolerate” to be able to use this type.
+	// e.g., cloud.provider.com/preemptible="true":NoSchedule
+	Taints []corev1.Taint `json:"taints,omitempty"`
+}
+
+// ResourceFlavorStatus defines the observed state of ResourceFlavor
```

Comment:
I think we night not ever need a status.

For those types, we don't even define a separate spec, just put the fields in the root struct https://github.com/kubernetes/api/blob/b9830ac37b4650e71097e16536fda6b752821122/storage/v1beta1/types.go#L612

Reply:
cool, removed.